iati/core/rulesets.py

```diff
@@ -1,45 +1,841 @@
+        """Initialise a `startswith` Rule."""
+        self.name = 'startswith'
+
+        super(RuleStartsWith, self).__init__(context, case)
+
+    def __str__(self):
+        """Return string stating what RuleStartsWith is checking."""
+        if len(self.paths) == 1:
+            return 'Each `{self.paths[0]}` within each `{self.context}` must start with the value present at `{self.start}`.'.format(**locals())
+        return 'Each instance of `{0}` within each `{self.context}` must start with the value present at `{self.start}`.'.format('` and `'.join(self.paths), **locals())
+
+    def _normalize_xpaths(self):
+        """Normalize xpaths by combining them with `context`."""
+        super(RuleStartsWith, self)._normalize_xpaths()
+
+        self.normalized_paths.append(self._normalize_xpath(self.start))
+
+    def _check_against_Rule(self, context_element):
+        """Assert that the prefixing text of all given `paths` starts with the text of `start`.
+
+        Args:
+            context_element (Element): An XML Element.
+
+        Returns:
+            bool: Return `True` when the `path` text starts with the text value of `start`.
+                  Return `False` when the `path` text does not start with the text value of `start`.
+
+        """
+        prefix = self._extract_text_from_element_or_attribute(context_element, self.start)[0]
+
+        for path in self.paths:
+            strings_to_check = self._extract_text_from_element_or_attribute(context_element, path)
+            for string_to_check in strings_to_check:
+                if not string_to_check.startswith(prefix):
+                    return False
+        return True
+
+
+class RuleSum(Rule):
+    """Representation of a Rule that checks that the values in given `path` attributes must sum to the given `sum` value."""
+
+    def __init__(self, context, case):
+        """Initialise a `sum` rule."""
+        self.name = 'sum'
+
+        super(RuleSum, self).__init__(context, case)
+
+    def __str__(self):
+        """Return string stating what RuleSum is checking."""
+        return 'Within each `{self.context}`, the sum of values matched at `{0}` must be `{self.sum}`.'.format('` and `'.join(self.paths), **locals())
+
+    def _check_against_Rule(self, context_element):
+        """Assert that the total of the values given in `paths` match the given `sum` value.
+
+        Args:
+            context_element (Element): An XML Element.
+
+        Returns:
+            bool: Return `True` when the `path` values total to the `sum` value.
+                  Return `False` when the `path` values do not total to the `sum` value.
+
+        """
+        unique_paths = set(self.paths)
+        values_in_context = list()
+
+        for path in unique_paths:
+            values_to_sum = self._extract_text_from_element_or_attribute(context_element, path)
+            for value in values_to_sum:
+                values_in_context.append(Decimal(value))
+
```

Comment:
Need to add a test that requires code along the lines of:

```python
if len(values_in_context) == 0:
  return True
```

to fix.

Reply:
done in separate pull request